pyqt-apps/siriushla/as_di_bpms/main.py

```diff
@@ -101,3 +107,32 @@ def setupui(self):
             self.layoutv.addWidget(grpbx)
             self.layoutv.addSpacing(20)
             self.layoutv.addStretch()
+
+    def _amcDetails(self, amc_number):
+        pbt = QPushButton('AFC Settings')
+        Window = create_window_from_widget(
+            utils.AFCAdvancedSettings, title=self.bpm+': AFC Settings')
+        util.connect_window(
+            pbt, Window, parent=self, prefix=self.prefix, display=self.bpm)
+
+        pv_list_fru = utils.AFCv3_1_PV_LIST['FRU']
+        grbx = CustomGroupBox('AFC Details', self)
+        grbx_lay = QHBoxLayout(grbx)
+        grbx_lay.setSpacing(20)
+
+        grbx_lay.addWidget(pbt)
+
+        crate = utils.device2crate(self.bpm)
+        cmd_button = SiriusPushButton(
+            self, label='Reset', icon=qta.icon('mdi.restart'),
+            init_channel=f"{crate}:{utils.DIS}-AMC-{amc_number}:{pv_list_fru['SoftRst']}"
+            )
+
+        label_mon = SiriusLabel(
+            self, f"{crate}:{utils.DIS}-AMC-{amc_number}:{pv_list_fru['SoftRstSts']}"
+            )
+
+        grbx_lay.addWidget(cmd_button)
+        grbx_lay.addWidget(label_mon)
+        self.layoutv.addWidget(grbx)
+        self.layoutv.addSpacing(20)
```

Comment:
I suggest these manipulations of the `self.layout` to be moved to the `self.setupui` method and this method to return the widget that is going to be inserted, this way it is clear in the `self.setup` all the manipulations to the layout itself.